ci/scripts/towncrier_automation.py

```diff
@@ -40,11 +44,12 @@ def parse_args(argv: Sequence[str] | None = None) -> Args:
         action="store_true",
     )
     args = parser.parse_args(argv, Args())
-    if len(version.Version(args.version).release) != 3:
-        raise ValueError(
-            f"Version argument {args.version} must contain major, minor, and patch version."
-        )
-    version.parse(args.version)  # validate
+    if (match := VERSION.fullmatch(args.version)) is None:
+        msg = f"Version argument {args.version} is not a valid version."
+        raise ValueError(msg)
+    if len(match["release"].split(".")) < 3:
+        msg = f"Version argument {args.version} must contain major, minor, and patch version."
+        raise ValueError(msg)
```

Comment:
Am I doing something wrong here?


```python
>>> import re
>>> from packaging import version
>>> from packaging.version import VERSION_PATTERN

>>> pattern = re.compile(VERSION_PATTERN, re.VERBOSE | re.IGNORECASE)
>>> pattern.fullmatch("1.0.0.0") # not None?
<re.Match object; span=(0, 7), match='1.0.0.0'>
>>> version.Version("1.0.0.0").release
(1, 0, 0, 0)
```

Reply:
So it seems like what I had before was right, no?

Reply:
huh, indeed! I think I confused that with the fact that `Version.patch` will be `0` no matter if it was supplied or not.

Will revert, sorry!